client/network/src/lib.rs

```diff
@@ -454,3 +454,92 @@ async fn wait_to_announce<Block: BlockT>(
 		);
 	}
 }
+
+pub async fn warp_sync_get<B, RCInterface>(
+	para_id: ParaId,
+	relay_chain_interface: RCInterface,
+	spawner: SpawnTaskHandle,
+) -> Result<oneshot::Receiver<<B as BlockT>::Header>, BoxedError>
+where
+	B: BlockT + 'static,
+	RCInterface: RelayChainInterface + 'static,
+{
+	let (sender, receiver) = oneshot::channel::<B::Header>();
+	spawner.spawn(
+		"cumulus-parachain-wait-for-target-block",
+		None,
+		async move {
+			tracing::debug!(
+				target: "cumulus-network",
+				"waiting for announce block in a background task...",
+			);
+			tracing::debug!(target: LOG_TARGET, "waiting for target block in a background task...",);
+			wait_for_target_block::<B, RCInterface>(sender, para_id, relay_chain_interface)
+				.await
+				.map_err(|e| {
+					tracing::error!(target: LOG_TARGET, "Unable to determine sync status. {}", e)
+				})
+				.unwrap();
+			tracing::debug!(target: LOG_TARGET, "target block reached",);
+		}
+		.boxed(),
+	);
+
+	return Ok(receiver)
+}
+
+async fn wait_for_target_block<B, RCInterface>(
+	sender: oneshot::Sender<<B as BlockT>::Header>,
+	para_id: ParaId,
+	relay_chain_interface: RCInterface,
+) -> Result<(), BoxedError>
+where
+	B: BlockT + 'static,
+	RCInterface: RelayChainInterface + 'static,
+{
+	match relay_chain_interface.import_notification_stream().await {
+		Ok(_) => loop {
```

Comment:
What are you doing here? Why are you creating a notification stream and don't use it? 

In general the logic of this function should be changed:

1. Create the notification stream.
2. In a loop wait until you have imported a block and `is_major_sync == false`.
3. Take the last finalized block and extract the parachain head from it.

Reply:
> What are you doing here? Why are you creating a notification stream and don't use it?

I thought we only needed the notification to inform us that the relay chain is synced and we can start the sync?